src/main/resources/uischema/scan.json

```diff
@@ -0,0 +1,102 @@
+{
+  "connectorName": "AI",
+  "operationName": "scan",
+  "title": "Scan a given image/document with AI",
+  "help": "<h1>Scan image/document with AI</h1> <b>Scan a given image/document with AI</b>",
+  "elements": [
+    {
+      "type": "attributeGroup",
+      "value": {
+        "groupName": "General",
+        "elements": [
+          {
+            "type": "attribute",
+            "value": {
+              "name": "description",
+              "displayName": "Description",
+              "inputType": "string",
+              "defaultValue": "Scan a given image/document with AI",
+              "required": "false",
+              "helpTip": "Give a description on the AI task"
+            }
+          },
+          {
+            "type": "attribute",
+            "value": {
+              "name": "configRef",
+              "displayName": "Connection",
+              "inputType": "connection",
+              "allowedConnectionTypes": [
+                "openai"
+              ],
+              "defaultType": "connection.openai",
+              "defaultValue": "",
+              "required": "true",
+              "helpTip": "Connection to be used"
+            }
+          },
+          {
+            "type": "attributeGroup",
+            "value": {
+              "groupName": "Request Parameters",
+              "elements": [
+                {
+                  "type": "attribute",
+                  "value": {
+                    "name": "fileName",
```

Comment:
Shall we rethink this approach as a user might wonder why they need to add both the file name and the content? If we are using the file name to get the type of file that is being used, maybe we need to let the user know that by showing this input only in scenarios where different type of files are being passed through the mediation flow.

Reply:
The reason behind expecting filename as a parameter is to identify the type of file being processed (either image/pdf). The research is going on to identify a general approach in identifying file types when we are using different mechanisms such as receive file as an attachment/ use of file connector/ VFS to read the file content. Will come up with a better approach after the research.